graphql_compiler/compiler/filters.py

```diff
@@ -949,7 +961,9 @@ def process_filter_directive(filter_operation_info, location, context):
 
     context['metadata'].record_filter_info(
         location,
-        FilterInfo(fields=fields, op_name=op_name, args=tuple(operator_params))
+        FilterInfo(fields=fields,
+                   op_name=op_name,
+                   args=tuple(operator_params) if operator_params else None)
```

Comment:
It seems like setting this to `None` is causing a lot of edge case issues in other code. Can we instead make this be an empty tuple if `operator_params` is empty or `None`? I think that we avoid almost all of the `len(parameters) if parameters else 0` and similar logic.